### ISSUE TITLE
[question] exact marker center

### ISSUE THREAD

**Issue by Huemerj** [alicevision/CCTag#149]
I'm trying out your library and first of all, I have to say that the results are quite nice!

During my tests, one question appeared regarding the exact coordinates of the circle center.
I tested the library with the samples you provide. They are 1400x1400px with 3 black rings.
So I was curious what the expected output of the center detection on these basic images is? I would expect (700,700) as this is the center of the image (also according to the crosses on your samples). Am I right or are there other things to consider, like an image convention which would shift the result by 0.5 px in both directions?
I'm asking because the results for the detection on your samples consistently lie in the range of 699.0-699.5 for both coordinates. 
I'm thankful for any clarification on that!

**Comment by simogasp** [comment]
Which markers are you using? From the pdf files or did you generate them with the python script?
It is actually strange, it could also be the rendering of the image.
@lcalvet

**Comment by Huemerj** [comment]
![0001](https://user-images.githubusercontent.com/9680593/105690657-8de07880-5efc-11eb-90ef-52dd5d9992f1.png)

For this example image, I get 699.38 699.585 as a result of the "detection" application.

**Comment by Huemerj** [comment]
Are these coordinates (699.38,699.585) pixel coordinates? With (0,0) being the center of the upper left pixel, this would mean that (699,699) is the center of the 700th pixel in both directions. This would make sense, as 699.5 would lie in between the 700th and 701st pixel, which would correspond to the center of the image.

**Comment by simogasp** [comment]
Ah yes, silly me. The output always considers the first pixel on the top left as the origin (0,0), while if u use any image editor it's likely to be considered (1,1).

**Comment by OndrejIvancik** [comment]
Hello, I am encountering the same problem. If the center of the first pixel has coordinates [0.5, 0.5] I would expect the center of the cctag to be exactly around [700.0, 700.0] -- this is the position between 700th and 701st pixel.
I found the problem on my dataset with 80 images (each has cca 6 detections). I created the second dataset and rotate it by 180 degrees. I run the detector and compare the coordinates (with respect of needed subtraction with image size). Results on 499 detections:
mean(err_x)=1.165, sum(err_x)=581.367, std(err_x)=0.455
mean(err_y)=1.089, sum(err_y)=543.781, std(err_y)=0.488
where err_x,y is already the absolute value of the error.

After adding the half pixel for both x and y coordinate:
mean(err_x)=0.243, sum(err_x)=121.214, std(err_x)=0.485
mean(err_y)=0.198, sum(err_y)=98.961, std(err_y)=0.456

I attached plots for error for x-coordinate.
![half_pixel_corrected](https://user-images.githubusercontent.com/75783056/105713041-f2123500-5f1a-11eb-8bde-63944c5c93c9.png)
![half_pixel_uncorrected](https://user-images.githubusercontent.com/75783056/105713075-fa6a7000-5f1a-11eb-8052-b418e34f79bf.png)
